utils/emissions.go

```diff
@@ -0,0 +1,429 @@
+	var dailyEmissions []models.EmissionsResult
+	startDatef := strings.Split(startDate.String(), " ")[0]
+	endDatef := strings.Split(endDate.String(), " ")[0]
+	networkName, _ := GetNetworkName(rpc)
+	chainIDs, _ := AccesibleChains(config)
+	for _, chain := range chainIDs {
+		intensityFactor, err := CarbonIntensityFactor(chain, startDatef, endDatef, config)
+		if len(intensityFactor) > 0 && err == nil {
+			dailyEmissions = append(dailyEmissions, models.EmissionsResult{
+				Chain: chain,
+				Value: GetAvgEmissionsValue(intensityFactor),
+			})
+		}
+	}
+	if dailyEmissions == nil {
+		return models.Emissions{Name: fmt.Sprintf("Daily emissions %s", networkName), Value: []models.EmissionsResult{}}
+	}
+	return models.Emissions{Name: fmt.Sprintf("Daily emissions %s", networkName), Value: dailyEmissions}
+}
+
+func GetNetworkEmissions(startDate time.Time, endDate time.Time, config cfg.EndpointService, rpc string) (models.Emissions, error) {
+	startDatef := strings.Split(startDate.String(), " ")[0]
+	endDatef := strings.Split(endDate.String(), " ")[0]
+	monthsBetween := int(endDate.Month() - startDate.Month())
+	yearsBetween := endDate.Year() - startDate.Year()
+	networkName, _ := GetNetworkName(rpc)
+
+	emissions := models.Emissions{
+		Name: fmt.Sprintf("Network Emissions %s", networkName),
+	}
+
+	emissionsResult, errEmissions := network(columbusChainID, startDatef, endDatef, config)
+	if errEmissions != nil || emissionsResult == nil {
+		emissions.Value = []models.EmissionsResult{}
+		return emissions, errEmissions
+	}
+
+	switch {
+	// if the date range is greater than or equal to one month the values are averaged per month
+	case (monthsBetween >= 1 || monthsBetween < 0 || startDate.Year() == 1) && yearsBetween == 0:
+		emissionsPermonth, err := GetAvgEmissionsFilter(emissionsResult, false)
+		if err != nil {
+			emissions.Value = []models.EmissionsResult{}
+			return emissions, errEmissions
+		}
+		emissions.Filter = perMonth
+		emissions.Value = emissionsPermonth
+	// if the date range is greater than or equal to one year the values are averaged per year
+	case yearsBetween > 0:
+		emissionsPerYear, err := GetAvgEmissionsFilter(emissionsResult, true)
+		if err != nil {
+			emissions.Value = []models.EmissionsResult{}
+			return emissions, errEmissions
+		}
+		emissions.Filter = perYear
+		emissions.Value = emissionsPerYear
+	default:
+		emissions.Filter = perDay
+		emissions.Value = emissionsResult
+	}
+	return emissions, nil
+}
+
+func GetCountryEmissions(startDate time.Time, endDate time.Time, config cfg.EndpointService, rpc string) (models.Emissions, error) {
+	startDatef := strings.Split(startDate.String(), " ")[0]
+	endDatef := strings.Split(endDate.String(), " ")[0]
+	networkName, _ := GetNetworkName(rpc)
+	var (
+		emissionsInfo []*models.CountryEmissionsResult
+		err           error
+	)
+	for _, country := range countryIDs {
+		countryInfo, errCountry := Country(country, startDatef, endDatef, config)
```

Comment:
I would rename `errCountry` with `err` and  remove both the `var err error` statement above and the else clause bellow.